adagrams/game.py

```diff
@@ -1,13 +1,80 @@
 from random import randint
 
+LETTER_POOL = {
+    'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2, 'G': 3, 'H': 2,
+    'I': 9, 'J': 1, 'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8, 'P': 2,
+    'Q': 1, 'R': 6, 'S': 4, 'T': 6, 'U': 4, 'V': 2, 'W': 2, 'X': 1,
+    'Y': 2, 'Z': 1
+}
+
 def draw_letters():
-    pass
+
+    # A temporary pool built from LETTER_POOL counts
+    pool = []
+    for letter, quantity in LETTER_POOL.items():
+        pool += [letter] * quantity
+    
+    hand = []
+    for _ in range(10):
+        idx = randint(0, len(pool) - 1)     # pick a random index
+        hand.append(pool[idx])              # add the letter at that index of pool
+        pool.pop(idx)                       # remove it so it can't be drawn again
+    
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    hand = []
+
+    for item in letter_bank:
+        hand.append(item)
+
+    for char in word.upper():
+        if char in hand:
+            idx = hand.index(char)
+            hand.pop(idx)
+        else:
+            return False
```

Comment:
Instead of using `else` to return `False`, we could flip the condition by checking to see if a `char` is not in the `hand` and then return `False`. 

If the check to see that a `char` is not in the `hand` isn't a true condition, then you can move on to find the index and pop the value from the hand. 

This allows you to remove `else`, which matters because every line of code is another place where a bug can be created, it reduces the conditional complexity of your code, and can highlight to other devs that the the main logic of the loop is to find a letter and remove it. 

```suggestion
        if char not in hand:
            return False
        idx = hand.index(char)
        hand.pop(idx)
```

Reply:
I have an additional thought about how to approach this problem. The performance of both checking whether something is in a list, and the remove function depends on the length of the list we are looking at. If I want to find something in a list, I have to visit each element until I find it (line 32). If I want to remove something, I have to visit each element, find it and then remove it (line 34). After I remove the element, I've left an empty space in my list and I have to move the other elements to fill that empty space.

And since we are doing these operations in a loop, it compounds the cost.

To improve the performance, we could iterate over `letter_bank` once to build a dictionary holding a count of how many times each letter appears, for example:
```py
letter_frequencies = {"a": 2, "f": 1}
```

As we process each letter in the word, we still need to look it up in the dictionary and check the count, but looking for a key in a dictionary is more efficient in a dictionary than in a list. If we find the letter is in the dictionary then we can just update the count by `-= 1` instead of removing an element. If you find a letter that has a value 0, then you know that the word uses more letters than are available. 

I'm not asking to see a refactored version of the approach I'm explaining, but it would be good practice to try it out for yourself!